### DISCUSSION TITLE
How can I mock a GraphQL client generated by Strawberry Shake client?

### DISCUSSION THREAD

**Discussion by BusHero** [ChilliCream/hotchocolate#6527] [Q&A]
I have a service that consumes a GraphQL client generated by Strawberry Shake. I want to write some unit tests for this service. How can I do that?
I see 3 ways:
1. Mock generated interfaces using a mock library
2. Testing against the real GraphQL server
3. Using `StrawberryShake.Transport.InMemory`

Based on this [SO question](https://stackoverflow.com/questions/71125729/how-to-replace-strawberry-shake-websocket-client), I have a feeling that option 3 is the answer, but I'm unable to make it work. I also didn't find any examples on how to use In Memory transport. Source code and tests provided little help.

Any suggestions?

**Comment by sgabler** [comment]
I personally would go with mocking. I use NSubstitute and my code looks like this:

```csharp
        var apiClient = Substitute.For<IMyApiClient>();
        apiClient.ExchangeOauthCode.ExecuteAsync(Arg.Any<OAuthCodeExchangeInput>())
            .Returns(new OperationResult<IExchangeOauthCodeResult>(
                new ExchangeOauthCodeResult(new ExchangeOauthCode_ExchangeOAuthCode_OAuthCodeExchangeOutput(
                    null,
                    new ExchangeOauthCode_ExchangeOAuthCode_AccessToken_AccessToken(600, "the-access-token", "Bearer", null),
                    null)
                ),
                null,
                null!,
                null)
            );
```

Not super readable, unfortunately.
